### DISCUSSION TITLE
Error when creating a user via API [Sep 01]

### DISCUSSION THREAD

**Discussion by dstala** [nocodb/nocodb#3739] [Discord Threads]
> 🙋‍♂️ **Manh-Trung BUI**

**Comment by o1lab** [comment]
> 🙋‍♂️ **wkw**  
#### please share the project info and the request body  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### Node: **v18.7.0**   
#### Arch: **x64**   
#### Platform: **linux**   
#### Docker: **false**   
#### Database: **mysql2**   
#### ProjectOnRootDB: **false**   
#### RootDB: **sqlite3**   
#### PackageVersion: **0.92.4**  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### {   
####     "email": "dsi@example.com",   
####     "password": "Information2022!",   
####     "roles": {   
####         "owner": true   
####     }   
#### }  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### It's correct ?  
#  
> 🙋‍♂️ **wkw**  
#### api endpoint && header pls  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
####   
![image0](https://cdn.discordapp.com/attachments/1012335193379586118/1012347881056895128/unknown.png)  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### http://nocodb.vival.com/api/v1/auth/user/signin  
#  
> 🙋‍♂️ **wkw**  
#### you want to create a user but using signin api?  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### yes  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### so i can make a script for a list of user  
#  
> 🙋‍♂️ **wkw**  
#### signin is for authentication only.  
#  
> 🙋‍♂️ **wkw**  
#### i think u want sign up  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### sorry it's signup  
#  
> 🙋‍♂️ **wkw**  
#### so you're using the wrong API then  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### Cannot GET /api/v1/auth/user/signup  
#  
> 🙋‍♂️ **wkw**  
#### that would be POST  
#  
> 🙋‍♂️ **wkw**  
#### see here for reference   
#### https://all-apis.nocodb.com/#tag/Auth/operation/auth-signup  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### thanks, i'll try this  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### response = requests.request("POST", url, headers=headers, params=querystring, data=json.dumps(body))  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
####   
![image0](https://cdn.discordapp.com/attachments/1012335193379586118/1012356768594149517/unknown.png)  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### I use POST method but always get this message  
#  
> 🙋‍♂️ **wkw**  
#### hv u tried `requests.post `  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### The same  
#  
> 🙋‍♂️ **wkw**  
#### The response looks like u r sending GET  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### response = requests.post(url, headers=headers, params=querystring, data=body)   
####    
#### But i was sended Post method  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### other APIs worked fine  
#  
> 🙋‍♂️ **Manh-Trung BUI**  
#### except signin and signup  
#  

 [Autogenerated from discord](https://discord.com/channels/661905455894888490/1012335193379586118)  
- - - -  
#### Join [NocoDB's](http://www.nocodb.com/)  community  
[Discord](https://discord.gg/5RgZmkW) • [Twitter](https://twitter.com/nocodb) • [Reddit](https://www.reddit.com/r/NocoDB/) • [Docs](https://docs.nocodb.com/)
